pkg/k8s/pods/pod.go

```diff
@@ -295,16 +295,24 @@ func Restart(dev *model.Dev, c *kubernetes.Clientset) error {
 		return fmt.Errorf("failed to retrieve dev environment information")
 	}
 
+	found := false
 	for i := range pods.Items {
-		err := c.CoreV1().Pods(dev.Namespace).Delete(pods.Items[i].Name, &metav1.DeleteOptions{GracePeriodSeconds: &devTerminationGracePeriodSeconds})
-		if err != nil {
-			if strings.Contains(err.Error(), "not found") {
-				return nil
+
+		if sn == "" || strings.HasPrefix(pods.Items[i].Name, fmt.Sprintf("%s-", sn)) {
```

Comment:
I'd build the prefix string once before the loop, since it doesn't change.

Reply:
For the condition check, I think it's more readable if you check if the pod should be skipped, and do a `continue`, instead of having a longer body inside the `if`.

Reply:
Yeah that makes more sense!